File: minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        if(s.size() == 1)return 0;
        int ans1 = 0;
        for(int i = 0 ; i < s.size() ; i++){
            if(i%2 == s[i] - '0'){
                ans1++;
            }
        }
        int ans2 = 0;
        for(int i = 0 ; i < s.size() ; i++){
            if(i%2 != s[i] - '0'){
                ans2++;
            }
        }
        
        return min(ans1,ans2);
    }
};

File: minimum-changes-to-make-alternating-binary-string/minimum-changes-to-make-alternating-binary-string.go
func min(i,j int)int{
    if i < j {
        return i;
    }
    return j;
}
func minOperations(s string) int {
    ans1,ans2 := 0,0
    length := len(s)
    for i := 0 ; i < length ; i++ {
        if i%2 == int(s[i]) - int('0') {
            ans1++;
        }else{
            ans2++;
        }
    } 
    return min(ans1,ans2);
}